test/boost/secondary_index_test.cc

```diff
@@ -1529,26 +1550,315 @@ SEASTAR_TEST_CASE(test_token_order) {
+                { int32_type->decompose(3), int32_type->decompose(2), int32_type->decompose(1), int32_type->decompose(3), int32_type->decompose(1) },
+                { int32_type->decompose(3), int32_type->decompose(2), int32_type->decompose(2), int32_type->decompose(1), int32_type->decompose(1) }
+            });
+
+            msg = cquery_nofail(e, "SELECT pk1, pk2, ck1, ck2, v2 FROM t WHERE token(pk1, pk2) >= token(1, 1) AND token(pk1, pk2) <= token(3, 2) AND v2 = 1 AND ck1 = 1 ALLOW FILTERING");
+            assert_that(msg).is_rows().with_rows({
+                { int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1) },
+                { int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(3), int32_type->decompose(1) },
+                { int32_type->decompose(3), int32_type->decompose(2), int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1) },
+                { int32_type->decompose(3), int32_type->decompose(2), int32_type->decompose(1), int32_type->decompose(3), int32_type->decompose(1) },
+            });
+
+            msg = cquery_nofail(e, "SELECT pk1, pk2, ck1, ck2, v2 FROM t WHERE token(pk1, pk2) >= token(1, 1) AND token(pk1, pk2) <= token(3, 2) AND v2 = 1 AND ck1 = 1 AND ck2 = 1 ALLOW FILTERING");
+            assert_that(msg).is_rows().with_rows({
+                { int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1) },
+                { int32_type->decompose(3), int32_type->decompose(2), int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1) },
+            });
+
+            msg = cquery_nofail(e, "SELECT pk1, pk2, ck1, ck2, v2 FROM t WHERE token(pk1, pk2) >= token(1, 1) AND token(pk1, pk2) <= token(3, 2) AND v2 = 1 AND ck2 = 1 ALLOW FILTERING");
+            assert_that(msg).is_rows().with_rows({
+                { int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1) },
+                { int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(2), int32_type->decompose(1), int32_type->decompose(1) },
+                { int32_type->decompose(3), int32_type->decompose(2), int32_type->decompose(1), int32_type->decompose(1), int32_type->decompose(1) },
+                { int32_type->decompose(3), int32_type->decompose(2), int32_type->decompose(2), int32_type->decompose(1), int32_type->decompose(1) }
+            });
+        };
+
+        // Do tests for each column (excluding partition key columns - cannot mix partition column 
+        // restrictions and TOKEN restrictions). First run the query without index, then with global index,
+        // and finally with local index.
+        eventually(do_tests_ck1);
```

Comment:
Out of curiosity, what does `eventually` do here?  If we need those queries to happen right now, before the index is created below, isn't `eventually` foiling that?

Reply:
Sorry, I confused `eventually` with `defer`.  But I'm still curious why exceptions are expected from do_tests_ck1.

Reply:
No, but really, the first run doesn't have an index, so why does it need eventually() at all?

Reply:
I agree, `eventually()` is not needed here (for queries without an index). Removed `eventually` from those queries in amended commit.